src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Windows.cs

```diff
@@ -252,5 +260,34 @@ internal int GetFileType()
 
             return fileType;
         }
+
+        internal unsafe long GetFileLength()
```

Comment:
> FileLength property rather than GetFileLength method?

Properties might be seen as something that is cheap to use. Since we can't cache file length on Unix and not always on Windows, I would prefer to keep it as a method. I hope that the callers are going to recognize it as something that is not for free.

Reply:
Ok, it's not a big deal (but FileStream.Length is a property).